llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -7651,6 +7651,12 @@ bool AMDGPULegalizerInfo::legalizeIntrinsic(LegalizerHelper &Helper,
     return legalizeLaneOp(Helper, MI, IntrID);
   case Intrinsic::amdgcn_s_buffer_prefetch_data:
     return legalizeSBufferPrefetch(Helper, MI);
+  case Intrinsic::amdgcn_dead: {
+    for (const MachineOperand &Def : MI.defs())
+      B.buildUndef(Def);
```

Comment:
Should this be poison? Comment that it should be poison, we don't have a G_POISON yet